feat(React): Ownership component of user profile

brendansun93 opened the pull request:
**Scope**
This PR only changes User profile component of React-web. 

**Changes**
This PR adds existing UserOwnership component to UserDetail component and adds data to display coming from GraphQL search query.

Below screenshot is taken from visiting http://localhost:3000/user/urn:li:corpuser:datahub/ownership/dataset
![image](https://user-images.githubusercontent.com/78824220/110341583-7a92f380-7ff8-11eb-979c-c97d040dcbfa.png)

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

brendansun93 commented on datahub-web-react/src/app/entity/user/UserProfile.tsx:
```diff
@@ -19,14 +21,69 @@ export default function UserProfile() {
+
+    const ownershipResult: any = {};
+
+    ownershipResult[EntityType.Chart] = useGetSearchResultsQuery({
```
Since hook cannot be called inside any loop, not sure if there is a way in Typescript that can make it dynamic, i.e. looping through all enum values and get search query

gabe-lyons replied:
We'd like to add support for an `all` endpoint in the future. In the meantime, I'd recommend encapsulating this all in a 2nd hook `useGetEachEntitySearchResults` so that the UserProfile render function isn't so large

jjoyce0510 replied:
+1 to wrapping this

brendansun93 commented:
@jjoyce0510 @gabe-lyons @shirshanka Please review

gabe-lyons reviewed (commented):
Nice! Looks good, although i do think it would benefit from encapsulating the multiplex search into its own helper util

gabe-lyons commented:
@brendansun I also noticed that the preview card looks squashed in your screenshot, do you know why that is? Does it looks the same as the dataset preview in search?

brendansun93 commented:
> @brendansun I also noticed that the preview card looks squashed in your screenshot, do you know why that is? Does it looks the same as the dataset preview in search?

No, the search result page shows ok. I think it is because the left side menu?

gabe-lyons commented:
@brendansun I see- feel free to change the UserOwnership component as you need to make them look the same

jjoyce0510 commented:
It looks like there should be more padding / margin on the individual list items that are rendered within the ownership details component... Can we include in this PR?

brendansun93 commented:
> It looks like there should be more padding / margin on the individual list items that are rendered within the ownership details component... Can we include in this PR?

OK got it, I can do that, will let you guys know when I am done with it

brendansun93 commented:
@gabe-lyons @jjoyce0510 I have made requested changes, please take a look again; also, the screenshot is updated with some vertical margin to look better :smile:

gabe-lyons commented:
@brendansun looks better- can you apply some extra padding on the first element so it is spaced out from the header equally to the element below it?

jjoyce0510 commented on datahub-web-react/src/app/entity/user/UserOwnership.tsx:
```diff
@@ -35,7 +35,15 @@ export default ({ ownerships, entityPath }: Props) => {
-                    return entityRegistry.renderPreview(entityType, PreviewType.PREVIEW, item);
+                    return (
+                        <div
+                            style={{
```
Nit: can we extract out styles into a `const` variable at the top of the file?

brendansun93 replied:
Removed inline style

jjoyce0510 commented on datahub-web-react/src/utils/customGraphQL/useGetEachEntitySearchResults.ts:
```diff
@@ -0,0 +1,44 @@
+import { EntityType } from '../../types.generated';
+import { useGetSearchResultsQuery } from '../../graphql/search.generated';
+
+export function useGetEachEntitySearchResults(input: any) {
```
nice! 

very minor nit: 

can we call this "useGetAllEntitySearchResults"

brendansun93 replied:
Done!

jjoyce0510 commented on datahub-web-react/src/utils/customGraphQL/useGetEachEntitySearchResults.ts:
```diff
@@ -0,0 +1,44 @@
+    result[EntityType.Chart] = useGetSearchResultsQuery({
+        variables: {
+            input: {
+                type: EntityType.Chart,
```
can we pass a "start" and "count" variable into this method and add it here? That way we can control how many results we get for each entity easily. I will likely replace my current entity loading in the "all" tab with what you've added here :)

brendansun93 replied:
This is already handled by the `input` object passed in to keep it generic, i.e. if we pass an object with start and count, then this query will get it. Do we want to explicitly mark these 2 parameters?

brendansun93 replied:
Please see latest change, I have implemented the `input` using `Partial<SearchInput>` so that `start` and `count` are checked when compile

jjoyce0510 replied:
Thank you! Looks great.

brendansun93 commented:
> @brendansun looks better- can you apply some extra padding on the first element so it is spaced out from the header equally to the element below it?

I have changed the code of this page a bit because Ant component doesnt allow us to style the margin for first item or the entire `ul` html element. Therefore I moved the Title out of list header with a divider under it now. Please see the screenshot i have just uploaded to description. If there should be more margin, please let me know and it will be a easy fix :)

gabe-lyons commented:
@brendansun looks good to me 🚀

gabe-lyons commented on datahub-web-react/src/utils/customGraphQL/useGetAllEntitySearchResults.ts:
```diff
@@ -0,0 +1,44 @@
+import { EntityType } from '../../types.generated';
+import { useGetSearchResultsQuery } from '../../graphql/search.generated';
+
+export function useGetAllEntitySearchResults(input: any) {
```
we should be able to get this type from search.generated- you don't need to leave this as an any

gabe-lyons replied:
Specifically it should be `SearchInput` from `src/types.generated.ts`

gabe-lyons replied:
If you want to allow any subset of fields from the type you could consider using `Partial<SearchInput>` as the type

brendansun93 replied:
Done!

brendansun93 replied:
Thanks for this information about `Partial`

gabe-lyons commented on datahub-frontend/run/frontend.env:
```diff
@@ -1,5 +1,5 @@
 # Server Port
-PORT=9001
+PORT=9002
```
make sure to change this back to 9001 before merging!

brendansun93 replied:
Reverted, thanks for the reminder

shirshanka reviewed (approved):
LGTM! Thanks @brendansun93 for the contrib and @gabe-lyons for the review!

jjoyce0510 reviewed (commented):
LGTM. Thank you so much!
